File: src/main/com/stockcomp/repository/LeaderboardEntryRepository.java
package com.stockcomp.repository;


import com.stockcomp.domain.leaderboard.LeaderboardEntry;
import com.stockcomp.domain.user.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface LeaderboardEntryRepository extends JpaRepository<LeaderboardEntry, Long> {

    List<LeaderboardEntry> findAllByOrderByRanking();

    List<LeaderboardEntry> findAllByOrderByScore();

    LeaderboardEntry findByUser(User user);
}

File: src/main/com/stockcomp/leaderboard/LeaderboardEntryRepository.kt
package com.stockcomp.leaderboard

import com.stockcomp.leaderboard.entity.LeaderboardEntry
import com.stockcomp.user.domain.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface LeaderboardEntryRepository : JpaRepository<LeaderboardEntry, Long> {
    fun findAllByOrderByScore(): List<LeaderboardEntry>

    fun findByUser(user: User): LeaderboardEntry?
}